packages/gatsby/src/redux/nodes.js

```diff
@@ -82,3 +82,45 @@ exports.getNodeAndSavePathDependency = (id, path) => {
   createPageDependency({ path, nodeId: id })
   return node
 }
+
+exports.saveResolvedNodes = async (typeName, resolver, nodeTypeNames) => {
+  const nodes = store.getState().nodesByType.get(typeName)
+  const resolvedNodes = new Map()
+  if (nodes) {
+    for (const node of nodes.values()) {
+      const resolved = await resolver(node)
+      resolvedNodes.set(node.id, resolved)
+    }
+    store.dispatch({
```

Comment:
could you add comment why we do this differently here than we do with loki? if i understand correctly, with loki we update the collection with the resolved nodes, but here we use a separate `resolvedNodes` Map? Also, how are `nodeTypeNames` handled here?

Reply:
I didn't want to mutate nodes directly in Redux store, because I was worried there would be some kind of sideeffects of that.